### PR TITLE
Refactored code and added ability to find missing translations

### LINKED ISSUE #2: Find missing translations too
This shouldn't be too difficult since we have analyzed most of the necessary data already

### PR DESCRIPTION
**Pull request by ijlee2**
I came up with a solution to issue #2 by taking these actions:
- Redefined `analyzeFiles` method to return a **Map**, so that we can later display in which file(s) a missing key had appeared
- Renamed `findUnusedTranslations` to `findDifferenceInTranslations` for clarity
- Refactored `findDifferenceInTranslations` method so that we can use it to find both unused and missing translations

I tested the updated code on a production app that has ~2900 translation keys (one locale). It found 816 unused translations 😞and 16 missing translations 🙂.

Resolves #2

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -41,6 +43,21 @@ async function run() {
       console.log(`   - ${key} ${chalk.dim(`(used in ${generateFileList(files)})`)}`);
     }
   }
+  console.log();
+
+  console.log(`${step(4)} ⚙️   Checking for missing translations...`);
+  let missingTranslations = findDifferenceInTranslations(usedTranslationKeys, existingTranslationKeys, whitelist);
```
Comment (Turbo87) [inline index.js]: can you move this step above the `unusedTranslations` reporting? I would like to have all the processing done before we start reporting the final results.

Comment (ijlee2) [reply index.js]: Yep, good idea. I can do so in the next hour, thanks.

File: index.js [hunk 2]
```diff
@@ -14,24 +14,28 @@ const Glimmer = require('@glimmer/syntax');
...
+  console.log();
+
   let translationFiles = await findTranslationFiles(rootDir);
   let existingTranslationKeys = await analyzeTranslationFiles(rootDir, translationFiles);
-
   let whitelist = config.whitelist || [];
-  let unusedTranslations = findUnusedTranslations(usedTranslationKeys, existingTranslationKeys, whitelist);
 
-  console.log();
+  let unusedTranslations = findDifferenceInTranslations(existingTranslationKeys, usedTranslationKeys, whitelist);
```
Comment (ijlee2) [inline index.js]: @Turbo87 To confirm, is grouping lines 36-37 what you wanted?

After I started reordering code, I realized that you may have instead meant to ask me to show the missing translations first, then the unused translations.

Comment (Turbo87) [reply index.js]: > To confirm, is grouping lines 36-37 what you wanted?

yes + the `console.log(`${step(4)} ⚙️   Checking for missing translations...`);` output too

Comment (ijlee2) [reply index.js]: Cool, I see now what you wanted to see. Have updated the code.

**Comment by Turbo87** [comment]
sorry, didn't have time to try this out yet 😞

**Comment by Turbo87** [comment]
thanks! and sorry again for the long wait... 😞

**Pull request merged**